src/compiler/transformers/destructuring.ts

```diff
@@ -562,7 +562,7 @@ namespace ts {
                     );
                 }
                 else {
-                    propertyNames.push(createLiteral(propertyName));
+                    propertyNames.push(createStringLiteral(getTextOfIdentifierOrLiteral(propertyName)));

```

Comment:
That's what `createLiteral` does, except it also sets `textSourceNode` to try to preserve the source text. I think the bug is actually in emitter.ts in `getLiteralTextOfNode`. In that function we check `isIdentifier(textSourceNode)` to ensure we add quotes, but we really should be checking `isIdentifier(textSourceNode) || isNumericLiteral(textSourceNode)` instead, and possibly ensuring we use the canonical string for the numeric literal (from its `text` property), since `1.0` toStrings to `"1"`.

Reply:
@rbuckton Thanks for the feedback 👍. I was thinking about changing the `getLiteralTextOfNode` however decided to change part of code related only to `__rest` helper :confused:.